### ISSUE TITLE
Can i share my altered Version of your macro?

### ISSUE THREAD

**Issue by MrJauhiainen** [Shuggaloaf/Simple_NPC_Generator#6]
Hey i absolutely love your macro i use it a lot to help me make a random npc's or Even just add a little flavour to the existing characters,
after some time i wanted pretty much the same thing you had already made just with a little more detail so i altered your macro to add things like Motive,Demeanor,Activity.
i also played around with the layout of it a bit and addend heading thats bold and a few other thing i just enjoy for myself, 
and now im still changing things but i also made an more in-depth version i use for key characters so ive added quite a bit like background,Occupation,motive/goal,Personality,secret and boons to the list of things it generates for me i dont always use all the stuff but its nice having stuff to play with or to help me get ideas, 

so i was just wondering if its ok if i share what ive done with it with others?, 
and would you like a copy to see for yourself?

**Comment by DanSlage** [comment]
I am another big fan of this module.
@MrJauhiainen, did you ever hear back from @Suggaloaf?  I would be interested in seeing your changes.  The macro source code says you can use the code as long as you attribute @Shuggaloaf so it should be OK.  Do you save your key generated characters to a journal entry? If so, what module do you use to copy the chat entry.  The only Foundry method I could find was to export the entire chat.

**Comment by Shuggaloaf** [comment]
Hey @MrJauhiainen sorry for the late reply! For whatever reason I didn't get a notification for this issue (Thanks for your @mention @DanSlage)

I actually wouldn't mind seeing those changes! Some of them sound very interesting. I may incorporate several at a later date, with your permission. Some of the items mentioned _may_ fall outside the scope of what I was looking for with this module (a simple alternative to the more complex generators out there). Although perhaps I'll love it all and want to add it in! :)

(BTW You mention occupation - that _should_ already be displaying under the NPC's name. Let me know if you're not seeing that.) 

&nbsp;

What I'd like to suggest is to fork my module and add your changes there. Once you have it done I'd love to see what you've got and then decide where to go from there. 

If it ends up being more than I'd like to incorporate, or if you would rather just release your own macro/module (and as long as this never goes commercial), then all I'd ask is to please give me contribution credits in the following places: 
&nbsp;&nbsp;&nbsp;&nbsp;1) In your Github ReadMe file AND if you ever publish as a Foundry module, in the module description;
&nbsp;&nbsp;&nbsp;&nbsp;2) At the top of the macro code; and 
&nbsp;&nbsp;&nbsp;&nbsp;3) As a contributor on your Github repo (forking mine might automatically add me, but I'm not sure)
Just a little sentence is plenty. Hopefully that's all reasonable. :) 

Or, if you weren't planning on creating a Github repo or releasing this, just let me know what you were planning and we can go from there. 

Thanks for showing an interest in this little pet project of mine that grew from a personal macro. I'm looking forward to seeing your additions!

**Comment by Shuggaloaf** [comment]
> what module do you use to copy the chat entry

@DanSlage Oh and to answer this question, checkout the Adventure Log feature of [DF Chat Enhancements](https://foundryvtt.com/packages/df-chat-enhance#adventure-log). 

I do notice though that it says version "10+" so I'm not sure if it doesn't work with v11 (I'm not using v11 yet) or just didn't need to be updated. If it still works though, or if you're not on FVTT v11 yet, it's a fantastic mod.

**Comment by MrJauhiainen** [comment]
Hey I don't expect you to implement all of my changes or even any of them I have very much have changed stuff just because I prefer it but it is awesome to hear you wanna see the changes I have made, 
and feel free to use the changes it would be awesome if I manage to add something to this amazing macro =)

ok so ill send two versions first is just some basic changes I made, so just added a little and changed the layout a bit,
[Reworked minor.txt](https://github.com/Shuggaloaf/Simple_NPC_Generator/files/12591541/Reworked.minor.txt)

and the second I probably have more then I need added to it so I don' tend to use all that pops up i just like having stuff to inspire me, and I wanna mess with the contents some more so gonna add some more to the lists so there more randomness or even take some of it out and probably change the wording so I am still very much testing out to see what I do and don't use and/or what needs to change.

[Major change testing NPC.txt](https://github.com/Shuggaloaf/Simple_NPC_Generator/files/12591543/Major.change.testing.NPC.txt)

anyway I hope you like some of what I have done.

**Comment by Shuggaloaf** [comment]
OK I got you. I wasn't sure what you were looking to do. 

So what was your original thoughts when you said you wanted to share your version? Were you going to publish a macro module or did you just mean something like share on Reddit/Discord/etc.? 

I'll take a look at these within the next few days (things are a bit busy right now). I'm always interested to see new ideas! I'll get back to you soon but feel free to reach out again. :)

**Comment by Shuggaloaf** [comment]
Hey @MrJauhiainen just wanted to let you know I have been working on this and have incorporated a few things from your macro. (Once I get further along I'll give you more specifics). 

So at this time I'd ask you to hold off sharing anything until I can complete my update. After that, I have some ideas for some of your code that I didn't use and how that can be better incorporated to work with my macro. 

TL;DR hang in there, there's some good stuff coming!

**Comment by Shuggaloaf** [comment]
@MrJauhiainen I'm closing this up with the release of versions 4.0. 

For your contributions, you are now listed as a contributor for this mod. 🥇 

&nbsp; 

A little about what I used and didn't use...
Just to give you an idea of my vision for this mod, I developed SNG to generate simple, street-level NPCs quickly. This is mainly for those NPCs players meet on the street, in a tavern, or wherever that you probably won't use again. And the ones you do use again, you won't need elaborate backstories or info for them. 

The barkeep. The shopkeep. The NPC your player asks for directions but they *have* to know their name. I'm not saying one of those NPCs may not develop into an NPC with more depth, but at that point you are probably best served by other modules or by writing out an NPC to your specifications. 

<p><b>I think a great idea would be to put the items below that I did not use into your own 2nd macro.</b><br>
Your macro could then be ran from mine with a simple 1 line call. This way you can add what you want to your macro, as often as you want, while keeping everything separate.Best part of this would be that if I update my mod, you wouldn't have to go back and re-add your changes, only that 1 line call to your macro.<br>If interested let me know. I'd be happy to either do this for you based on what you gave me or help you create it.</p>

<details> 
  <summary>YOUR ITEMS THAT I USED</summary>
   <h3>"Category" Titles</h3>
   <p>I like the way you added the bold titles for each item. It makes it easier to spot what you're looking for quickly. Which is what this mod is all about. Adding this!</p>
<h3>Physical Characteristics</h3>
<p>I rolled a lot of these into my "descriptor" category (what you're using as Defining Feature). 
I prefer to not have a second category as to limit physical features, allowing the DM more creativity. 
It also helps avoid conflicts. For example the first NPC I rolled with your additions had all 3 of these "one eye", "uneven eyes" and "twinkling orange eyes".</p>
<h3>Professions</h3>
<p>Many were duplicates of what I had, just in different terms. Examples: your "boat builder" to my "shipwright" (you also had "boatwright"). Where I had "clergy member" you had several like "bishop, chaplin, etc."
Here are the ones I used:  Courtesan, Custodian, Milkmaid/lad, Monk, Sapper, Servant, Sign Maker, Steward / Majordomo, Swineherd, Watchman, Wheelwright</p>
<h3>Activity</h3>
<p>I like this and think I may add some of your list. There's only a few I think may not fit depending on the situation, but most of what you have here I may use. </p>
</details>

<details> 
  <summary>YOUR ITEMS I DID NOT USE</summary>
    <p> (and why, just to give you context)</p>
    <p><b>Note: these are the items I'm suggesting above that we make as an add-on macro. </b></p>

<h3>Goals</h3>
<p>I went back and forth but ultimately, for the quick, generic NPCs my generator is designed for, I didn't go with this.</p>
</h3>Motive</h3>
<p>Did not add for reasons similar to Goals.</p>
</h3>Personality</h3>
<p>Did not use, same reasons as above.</p>
</h3>Backgrounds</h3>
<p>While the idea of having a background isn't a bad one, I ended up leaving this out for the same reasons as above. It's not something I think will be needed for _most_ random NPCs. This also allows the DM more creativity as they're not limited to a specific background.</p>
</h3>Secrets</h3>
<p>I did not add any at this time. I do like the idea of having an optional secrets category. **I will likely add this in the future**, but it is going to be 95% mundane stuff though since this is for random street-level NPCs.
In other words, to take a few examples of yours: More of this: "Knows the best fishing spot" and very little of this: "Is an assassin/spy"</p>
</h3>Boon</h3>
<p>Looking at the list of choices you had, it feels like these would be unlikely for random NPCs. I could see these being useful for a more involved/"written" NPC, but not so much for the random "meet on the street" type NPCs this generator is designed (unless your PC party are well-known heroes).</p>
</details>
